### PR TITLE
Integrate `arethetypeswrong`

### LINKED ISSUE #38: Verify our exports with arethetypeswrong
https://arethetypeswrong.github.io/ verifies type exports of NPM packages with the version resolution algorithms implemented in TypeScript. 

It's available as a CLI, and we should use it here to guard against regressions. For reference, see https://github.com/connectrpc/connect-es/pull/838.

### PR DESCRIPTION
**Pull request by smaye81**
Fixes #38 

This adds the `arethetypeswrong` CLI to our codebase and also runs it in CI.

### CODE REVIEW

File: turbo.json [hunk 1]
```diff
@@ -20,6 +20,9 @@
     },
     "lint": {
       "dependsOn": ["format", "^build"]
+    },
+    "attw": {
+      "dependsOn": ["^build"]
```
Comment (timostamm) [inline turbo.json:25]: This is incorrect, see https://github.com/connectrpc/connect-query-es/pull/184#discussion_r1341713169

Comment (smaye81) [reply turbo.json]: Yep, fixed.

**Review by timostamm — APPROVED** [review]
LGTM. Can you merge this into https://github.com/connectrpc/connect-playwright-es/pull/37 ?

**Comment by smaye81** [comment]
> LGTM. Can you merge this into #37 ?

Yep, will do.

**Pull request merged**